docker-compose-rule-core/src/main/java/com/palantir/docker/compose/execution/DockerCommandLocator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.docker.compose.execution;
+
+import static java.util.Collections.singletonList;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.SystemUtils;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class DockerCommandLocator {
+
+    protected abstract String command();
+
+    @Value.Default
+    protected boolean isWindows() {
+        return SystemUtils.IS_OS_WINDOWS;
+    }
+
+    @Value.Derived
+    protected String executableName() {
+        if (isWindows()) {
+            return command() + ".exe";
+        }
+        return command();
+    }
+
+    @Nullable
+    protected abstract String locationOverride();
+
+    @Value.Derived
+    protected List<Path> searchLocations() {
+        if (locationOverride() == null) {
+            return DockerCommandLocations.pathLocations();
+        }
+        return singletonList(Paths.get(locationOverride()));
+    }
+
+    public String getLocation() {
```

Comment:
Previously DOCKER_COMPOSE_LOCATION would be the path to the binary, e.g. `/path/to/docker-compose`, but the logic here requires it to point to the directory in which to find docker-compose, i.e. `/path/to`. I think the previous behaviour is more obvious/expected.

You could remove the `searchLocations` method, and do the check for the location override here instead, e..g

```
if (locationOverride() == null) {
  return DockerCommandLocations.pathLocations().stream()....;
} else {
  return locationOverride();
}
```

It might even be worth moving `DockerCommandLocations.pathLocations()` to this class, but that's just a design thing.